pkgs/development/libraries/wolfssl/default.nix

```diff
@@ -6,73 +6,102 @@
 , util-linux
 , openssl
 }:
-
-stdenv.mkDerivation rec {
-  pname = "wolfssl";
-  version = "5.6.3";
-
-  src = fetchFromGitHub {
+let
+  defaultVersion = "5.6.3";
+  defaultSrc = fetchFromGitHub {
     owner = "wolfSSL";
     repo = "wolfssl";
-    rev = "refs/tags/v${version}-stable";
+    rev = "refs/tags/v${defaultVersion}-stable";
     hash = "sha256-UN4zs+Rxh/bsLD1BQA+f1YN/UOJ6OB2HduhoetEp10Y=";
   };
 
-  postPatch = ''
-    patchShebangs ./scripts
-    # ocsp tests require network access
-    sed -i -e '/ocsp\.test/d' -e '/ocsp-stapling\.test/d' scripts/include.am
-    # ensure test detects musl-based systems too
-    substituteInPlace scripts/ocsp-stapling2.test \
-      --replace '"linux-gnu"' '"linux-"'
-  '';
+  buildWolfSSL =
+    { version ? defaultVersion
+    , src ? defaultSrc
+    , variant
+    , defaultFlags ? true
+    , extraOptions ? [ ]
+    , extraFlags ? [ ]
+    }: stdenv.mkDerivation rec {
+      pname = "wolfssl-${variant}";
+
+      inherit version src;
+
+      postPatch = ''
+        patchShebangs ./scripts
+        # ocsp tests require network access
+        sed -i -e '/ocsp\.test/d' -e '/ocsp-stapling\.test/d' scripts/include.am
+        # ensure test detects musl-based systems too
+        substituteInPlace scripts/ocsp-stapling2.test \
+          --replace '"linux-gnu"' '"linux-"'
+      '';
+
+      configureFlags = (map (o: "--enable-${o}") ([
+        "reproducible-build"
+      ] ++ lib.optionals (defaultFlags) [
+        "${variant}"
+      ] ++ lib.optionals (defaultFlags && stdenv.hostPlatform.isx86) [
+        # Enable AVX/AVX2/AES-NI instructions, gated by runtime detection via CPUID.
+        "intelasm"
+        "aesni"
+      ] ++ lib.optionals (defaultFlags && stdenv.isAarch64 && stdenv.isDarwin) [
```

Comment:
Wouldn't this apply to aarch64-linux, too?

Reply:
Yes, but only on some machines.
IIRC the aarch64-linux target is `armv8+a`, so the crypto extensions aren't there. As there is no detection, we can't assume they exist.